File: src/dotnet/Taml/Taml/TamlDocument.cs
using System.Collections.Generic;

namespace TAML
{

	/// <summary>
	/// A representation of a TAML document
	/// </summary>
	public class TamlDocument
	{

		/// <summary>
		/// The collection of data stored in the document
		/// </summary>
		/// <typeparam name="TamlKeyValuePair">The data represented in the document</typeparam>
		/// <returns></returns>
		public List<TamlKeyValuePair> KeyValuePairs { get; set; } = new List<TamlKeyValuePair>();

		/// <summary>
		/// Instructions for the parser that were identified in the document
		/// </summary>
		/// <typeparam name="TamlKeyValuePair"></typeparam>
		/// <returns>The instruction keys and their configured values</returns>
		public IEnumerable<TamlKeyValuePair> ProcessorDirectives { get; set; } = new List<TamlKeyValuePair>();

	}

}

File: src/JVM/taml4j-parser/src/main/java/com/taml4j/TamlDocument.java
package com.taml4j;

import java.util.ArrayList;
import java.util.List;

/**
 * @author Luca Taddeo - Lucalas
 */
public class TamlDocument {
	/**
	 * The collection of data stored in the document
	 */
	private List<TamlKeyValuePair> keyValuePairs = new ArrayList<>();

	/**
	 * Instructions for the parser that were identified in the document
	 */
	private TamlMap processorDirectives = new TamlMap();

	public void setProcessorDirectives(TamlMap processorDirectives) {
		this.processorDirectives = processorDirectives;
	}

	public List<TamlKeyValuePair> getKeyValuePairs() {
		return keyValuePairs;
	}

	public TamlMap getProcessorDirectives() {
		return processorDirectives;
	}
}
